src/views/home.tsx

```diff
@@ -1,32 +1,37 @@
-import React from 'react';
-import List from '../components/list.js';
-import Header from '../components/header';
+import React from "react";
+import { List } from "../components/list";
+import Header from "../components/header";
+import { useDispatch, useSelector } from "react-redux";
+import { FalconRootState } from "../store/rootReducer";
+import { getSongList } from "../store/list/effects";
+import { FeedListType } from "../store/list/types";
+
+export const Home = () => {
+  let dispatch = useDispatch();
+  let { frequent, latest } = useSelector(
+    (state: FalconRootState) => state.feed
+  );
+
+  React.useEffect(() => {
+    if (!frequent.posts.length) {
+      dispatch(getSongList(FeedListType.frequent, 0, 25));
+    }
+  }, [frequent.posts.length, dispatch]);
+
+  React.useEffect(() => {
```

Comment:
Can we combine this useEffect with the one above this? with [frequent.posts.length, latest.posts.length, dispatch] as the second arguments?

Reply:
Um, no we shouldn't. Let's say I write it as:
```js
React.useEffect(() => {
  if (!frequent.posts.length) {
    dispatch(getSongList(FeedListType.frequent, 0, 25));
  }
  if (!latest.posts.length) {
    dispatch(getSongList(FeedListType.latest, 0, 25));
  }
}, [frequent.posts.length, latest.posts.length, dispatch]);
```

Here's what will happen:
1. Store has no 'frequent' posts so an action for the 'frequent' list gets dispatched
2. Store has no 'latest' posts so an action for the 'latest' list gets dispatched
3. But even if you consider both network calls to take the same time, frequent list will be updated, causing this `useEffect` to be triggered again (because `frequent.posts.length` has changed.
4. This time, ` if (!frequent.posts.length)` will not be satisfied but because the 'latest' list hasn't updated yet, ` if (!latest.posts.length)` condition will satisfy, dispatching the action for 'latest' list again, which is not desirable.

Reply:
Thanks for explaining the async problem with combining them :+1:!